repos/system_upgrade/el7toel8/actors/initramfs/targetinitramfsgenerator/libraries/targetinitramfsgenerator.py

```diff
@@ -0,0 +1,78 @@
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import InitrdIncludes, InstalledTargetKernelVersion, TargetInitramfsTasks
+from leapp.utils.deprecation import suppress_deprecation
+
+
+DRACUT_DIR = '/usr/lib/dracut/modules.d/'
+
+
+def copy_dracut_modules(modules):
+    """
+    Copy every dracut module with specified path into the expected directory.
+
+    original content is overwritten if exists
+    """
+    # FIXME: use just python functions instead of shell cmds
+    for module in modules:
+        if not module.module_path:
+            continue
+        try:
+            # context.copytree_to(module.module_path, os.path.join(DRACUT_DIR, os.path.basename(module.module_path)))
+            run(['cp', '-f', '-a', module.module_path, DRACUT_DIR])
+        except CalledProcessError as e:
+            api.current_logger().error('Failed to copy dracut module "{name}" from "{source}" to "{target}"'.format(
+                name=module.name, source=module.module_path, target=DRACUT_DIR), exc_info=True)
+            # FIXME: really do we want to raise the error and stop execution completely??....
+            raise StopActorExecutionError(
+                message='Failed to install dracut modules required in the target initramfs. Error: {}'.format(str(e))
+            )
+
+
+@suppress_deprecation(InitrdIncludes)
+def _get_files():
+    files = {f for i in api.consume(InitrdIncludes) for f in i.files}
+    files.update([f for i in api.consume(TargetInitramfsTasks) for f in i.include_files])
+    return files
+
+
+def _get_modules():
+    # NOTE: See the comments in intramdiskgenerator about the handling
```

Comment:
I would not be afraid of duplicating info of it is important here, rather then pointing to another place.

Reply:
Good catch. The note is nowadays wrong regarding the new actor that is checking possible conflicts in *initramfs tasks*. The note is updated